gtkhash 1.5 build error against nettle 4

gtkhash 1.5 FTBFS against nettle 4 with
```
hash-lib-nettle.c: In function 'gtkhash_hash_lib_nettle_finish':
hash-lib-nettle.c:106:47: error: passing argument 2 of '((struct hash_lib_nettle_s *)func->lib_data)->meta->digest' makes pointer from integer without a cast [-Wint-conversion]
  106 |         LIB_DATA->meta->digest(LIB_DATA->ctx, *size, digest);
      |                                               ^~~~~
      |                                               |
      |                                               size_t {aka long unsigned int}
hash-lib-nettle.c:106:47: note: expected 'uint8_t *' {aka 'unsigned char *'} but argument is of type 'size_t' {aka 'long unsigned int'}
hash-lib-nettle.c:36:19: error: too many arguments to function '((struct hash_lib_nettle_s *)func->lib_data)->meta->digest'; expected 2, have 3
```

--- a/src/hash/hash-lib-nettle.c
+++ b/src/hash/hash-lib-nettle.c
@@ -27,6 +27,7 @@
 #include <glib.h>
 #include <nettle/nettle-meta.h>
 #include <nettle/sha3.h>
+#include <nettle/version.h>
 
 #include "hash-lib.h"
 #include "hash-func.h"
@@ -103,7 +104,11 @@ uint8_t *gtkhash_hash_lib_nettle_finish(struct hash_func_s *func,
 	*size = LIB_DATA->meta->digest_size;
 	uint8_t *digest = g_malloc(*size);
 
+#if NETTLE_VERSION_MAJOR >= 4
+	LIB_DATA->meta->digest(LIB_DATA->ctx, digest);
+#else
 	LIB_DATA->meta->digest(LIB_DATA->ctx, *size, digest);
+#endif
 
 	g_free(LIB_DATA->ctx);
 	g_free(LIB_DATA);

